moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -32,132 +32,338 @@
+    // Time only the jacobian computation, not the forward kinematics.
+    st.PauseTiming();
+    kinematic_state.setToRandomPositions(jmg, rng);
+    kinematic_state.updateLinkTransforms();
+    st.ResumeTiming();
+    kinematic_state.getJacobian(jmg);
   }
 }
 
-int main(int argc, char** argv)
+static void KdlJacobian(benchmark::State& st)
 {
-  testing::InitGoogleTest(&argc, argv);
-  return RUN_ALL_TESTS();
+  const moveit::core::RobotModelPtr& robot_model = moveit::core::loadTestingRobotModel(TEST_ROBOT);
+
+  // Make sure the group exists, otherwise exit early with an error.
+  if (!robot_model->hasJointModelGroup(TEST_GROUP))
+  {
+    st.SkipWithError("The planning group doesn't exist.");
+    return;
+  }
+
+  // Robot state.
+  moveit::core::RobotState kinematic_state(robot_model);
+  const moveit::core::JointModelGroup* jmg = kinematic_state.getJointModelGroup(TEST_GROUP);
+
+  // Provide our own random number generator to setToRandomPositions to get a deterministic sequence of joint
+  // configurations.
+  random_numbers::RandomNumberGenerator rng(0);
+
+  KDL::Tree kdl_tree;
+  if (!kdl_parser::treeFromUrdfModel(*robot_model->getURDF(), kdl_tree))
+  {
+    st.SkipWithError("Can't create KDL tree.");
+    return;
+  }
+
+  KDL::Chain kdl_chain;
+  if (!kdl_tree.getChain(jmg->getJointModels().front()->getParentLinkModel()->getName(),
+                         jmg->getLinkModelNames().back(), kdl_chain))
+  {
+    st.SkipWithError("Can't create KDL Chain.");
+    return;
+  }
+
+  KDL::ChainJntToJacSolver jacobian_solver(kdl_chain);
+
+  for (auto _ : st)
+  {
+    st.PauseTiming();
+    kinematic_state.setToRandomPositions(jmg, rng);
+    kinematic_state.updateLinkTransforms();
+    KDL::Jacobian jacobian(kdl_chain.getNrOfJoints());
+    KDL::JntArray kdl_q;
+    kdl_q.resize(kdl_chain.getNrOfJoints());
+    kinematic_state.copyJointGroupPositions(jmg, &kdl_q.data[0]);
+    st.ResumeTiming();
+    jacobian_solver.JntToJac(kdl_q, jacobian);
+  }
 }
+
+BENCHMARK(MultiplyAffineTimesMatrix)->Arg(1e7)->Unit(benchmark::kMillisecond);
+BENCHMARK(MultiplyMatrixTimesMatrix)->Arg(1e7)->Unit(benchmark::kMillisecond);
+BENCHMARK(MultiplyIsometryTimesIsometry)->Arg(1e7)->Unit(benchmark::kMillisecond);
+
+BENCHMARK(InverseIsometry3d)->Arg(1e7)->Unit(benchmark::kMillisecond);
+BENCHMARK(InverseAffineIsometry)->Arg(1e7)->Unit(benchmark::kMillisecond);
+BENCHMARK(InverseAffine)->Arg(1e7)->Unit(benchmark::kMillisecond);
+BENCHMARK(InverseMatrix4d)->Arg(1e7)->Unit(benchmark::kMillisecond);
+
+BENCHMARK(RobotStateConstruct)->RangeMultiplier(10)->Range(100, 10000)->Unit(benchmark::kMillisecond);
+BENCHMARK(RobotStateCopy)->RangeMultiplier(10)->Range(100, 10000)->Unit(benchmark::kMillisecond);
+BENCHMARK(RobotStateUpdate)->RangeMultiplier(10)->Range(10, 1000)->Unit(benchmark::kMillisecond);
+BENCHMARK(RobotStateForwardKinematics)->RangeMultiplier(10)->Range(10, 1000)->Unit(benchmark::kMillisecond);
+
```

Comment:
I feel like a lot of these could be made into `constexpr`s to clarify intent and reduce copypasta errors if/when changes must be made